### PR TITLE
added 01_05_homework (promises tasks)

### PR DESCRIPTION
**Pull request by ArtstiomEzepchick**
Smilejke, please check my 01_05_homework (promises tasks)

### CODE REVIEW

File: 01_05/promises.js [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+function delay(ms) {
+  // ваш код
+}
+
+delay(3000).then(() => alert('выполнилось через 3 секунды'));
+
+// МОЁ РЕШЕНИЕ НИЖЕ
+
+function newDelay(ms) {
+    const delayPromise = new Promise((resolve) => {
```
Comment (smilejke) [inline 01_05/promises.js:31]: can be done in one string (return new Promise...)

File: 01_05/promises.js [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    div.className = 'circle';
+    document.body.append(div);
+
+    const showCirclePromise = new Promise(resolve => {
+        setTimeout(() => {
+            div.style.width = radius * 2 + 'px';
+            div.style.height = radius * 2 + 'px';
+      
+            div.addEventListener('transitionend', function handler() {
+              div.removeEventListener('transitionend', handler);
```
Comment (smilejke) [inline 01_05/promises.js:105]: unsafe code, but we will learn it next class, so no need to change

File: 01_05/promises.js [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+// Являются ли фрагменты кода ниже эквивалентными? Другими словами, ведут ли они себя одинаково во всех обстоятельствах, для 
+// всех переданных им обработчиков?
+
+promise.then(f1).catch(f2);
+
+// Против:
+
+promise.then(f1, f2);
+
+// ОТВЕТ: Не эквивалентны, т.к. в первом случае, если будет ошибка - она обработается с помощью catch, а во втором случае - ошибка не будет обработана
```
Comment (smilejke) [inline 01_05/promises.js:125]: wrong answer, see https://learn.javascript.ru/promise-basics#potrebiteli-then-catch-finally

File: 01_05/promises.js [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+      }
+    })
+}
+
+loadJson('no-such-user.json')
+  .catch(console.log); // Error: 404
+
+// МОЁ РЕШЕНИЕ НИЖЕ
+
+async function loadNewJson(url) {
```
Comment (smilejke) [inline 01_05/promises.js:163]: Wrong solution.
You need to use try/catch in the function and then just call it with `loadNewJson('no-such-user.json')`
Refactor it

File: 01_05/promises.js [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+class HttpError extends Error {
+  constructor(response) {
+    super(`${response.status} for ${response.url}`);
+    this.name = 'HttpError';
+    this.response = response;
+  }
+}
+
+async function loadJson(url) {
+    const response = await fetch(url);
```
Comment (smilejke) [inline 01_05/promises.js:231]: the same as prev task. Try/catch here.

File: 01_05/promises.js [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+
+    response.status == 200 ? response.json() : new HttpError(response);
```
Comment (smilejke) [inline 01_05/promises.js:233]: even with your wrong approach, this code is missing `return` statement

File: 01_05/promises.js [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+
+loadJson('no-such-user.json')
+  .catch(console.log); // Error: 404
+
+// МОЁ РЕШЕНИЕ НИЖЕ
+
+async function loadNewJson(url) {
+    const response = await fetch(url);
+
+    response.status == 200 ? response.json() : new Error(response.status);
```
Comment (smilejke) [inline 01_05/promises.js:166]: even with your wrong approach, this code is missing `return` statement

File: 01_05/promises.js [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+};
+
+async function demoGithubUser() {
+    const gitHubUser;
+
+    while(true) {
+        let name = prompt("Введите логин?", "iliakan");
+
+        try {
+            gitHubUser = await loadJson(`https://api.github.com/users/${name}`);
```
Comment (smilejke) [inline 01_05/promises.js:243]: `gitHubUser` is declared with `const` keyword and won't be changed

File: 01_05/promises.js [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+async function loadJson(url) {
+    const response = await fetch(url);
+
+    response.status == 200 ? response.json() : new HttpError(response);
+};
+
+async function demoGithubUser() {
+    const gitHubUser;
+
+    while(true) {
```
Comment (smilejke) [inline 01_05/promises.js:239]: this statement lead to infinite loop

File: 01_05/promises.js [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+  }
+}
+
+async function loadJson(url) {
+    const response = await fetch(url);
+
+    response.status == 200 ? response.json() : new HttpError(response);
+};
+
+async function demoGithubUser() {
```
Comment (smilejke) [inline 01_05/promises.js:236]: wrong solution, rework

File: 01_05/promises.js [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+
+    while(true) {
+        let name = prompt("Введите логин?", "iliakan");
+
+        try {
+            gitHubUser = await loadJson(`https://api.github.com/users/${name}`);
+            console.log(`Полное имя: ${gitHubUser.name}.`);
+            break;
+        } catch {
+            err instanceof HttpError && err.response.status == 404 ?
```
Comment (smilejke) [inline 01_05/promises.js:247]: we're not using `==`

**Review by smilejke — CHANGES_REQUESTED** [review]

File: 01_05/promises.js (hunk 2 above)
Comment (ArtstiomEzepchick) [reply 01_05/promises.js:105]: got it

File: 01_05/promises.js (hunk 1 above)
Comment (ArtstiomEzepchick) [reply 01_05/promises.js:31]: changed

File: 01_05/promises.js (hunk 3 above)
Comment (ArtstiomEzepchick) [reply 01_05/promises.js:125]: corrected my answer

File: 01_05/promises.js (hunk 11 above)
Comment (ArtstiomEzepchick) [reply 01_05/promises.js:247]: changed

File: 01_05/promises.js (hunk 9 above)
Comment (ArtstiomEzepchick) [reply 01_05/promises.js:239]: Why infinite? I've used break-rule in try/catch construction. It doesn't help to quit cycle?

File: 01_05/promises.js (hunk 8 above)
Comment (ArtstiomEzepchick) [reply 01_05/promises.js:243]: changed from "const" to "let"

File: 01_05/promises.js (hunk 4 above)
Comment (ArtstiomEzepchick) [reply 01_05/promises.js:163]: refactored

File: 01_05/promises.js (hunk 5 above)
Comment (ArtstiomEzepchick) [reply 01_05/promises.js:231]: changed

File: 01_05/promises.js [hunk 12]
```diff
@@ -122,7 +121,8 @@ promise.then(f1).catch(f2);
 
 promise.then(f1, f2);
 
-// ОТВЕТ: Не эквивалентны, т.к. в первом случае, если будет ошибка - она обработается с помощью catch, а во втором случае - ошибка не будет обработана
+// ОТВЕТ: Не эквивалентны, т.к. в первом случае, если будет ошибка - она обработается с помощью catch, а во втором случае нету
```
Comment (smilejke) [inline 01_05/promises.js:124]: will discuss during next class

**Review by smilejke — APPROVED** [review]

**Pull request merged**